### ISSUE TITLE
Installation issue

### ISSUE THREAD

**Issue by vikaswadhwa-dgtl-factory** [Martin-Nyaga/fast_statistics#2]
Hello,

Looking forward to using this gem! Running into an installation problem, wondering if this is due to ruby version or something I'm missing:

```
Installing fast_statistics 0.2.0 with native extensions
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

current directory:
~/.gem/ruby/2.5.5/gems/fast_statistics-0.2.0/ext/fast_statistics
~/.rubies/ruby-2.5.5/bin/ruby -r
./siteconf20210701-51209-f1beda.rb extconf.rb
checking for xmmintrin.h... yes
creating Makefile

current directory:
~/.gem/ruby/2.5.5/gems/fast_statistics-0.2.0/ext/fast_statistics
make "DESTDIR=" clean

current directory:
~/.gem/ruby/2.5.5/gems/fast_statistics-0.2.0/ext/fast_statistics
make "DESTDIR="
compiling array_2d.cpp
array_2d.cpp:20:12: warning: implicit conversion loses integer precision: 'long'
to 'int' [-Wshorten-64-to-32]
    cols = rb_array_len(arrays);
         ~ ^~~~~~~~~~~~~~~~~~~~
array_2d.cpp:21:12: warning: implicit conversion loses integer precision: 'long'
to 'int' [-Wshorten-64-to-32]
    rows = rb_array_len(rb_ary_entry(arrays, 0));
         ~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2 warnings generated.
compiling fast_statistics.cpp
fast_statistics.cpp:55:40: error: expected expression
static rb_data_type_t dfloat_wrapper = [] {
                                       ^
1 error generated.
make: *** [fast_statistics.o] Error 1

make failed, exit code 2

Gem files will remain installed in
~/.gem/ruby/2.5.5/gems/fast_statistics-0.2.0 for inspection.
Results logged to
~/.gem/ruby/2.5.5/extensions/x86_64-darwin-19/2.5.0-static/fast_statistics-0.2.0/gem_make.out

An error occurred while installing fast_statistics (0.2.0), and Bundler
cannot continue.
Make sure that `gem install fast_statistics -v '0.2.0' --source
'https://rubygems.org/'` succeeds before bundling.
```

**Comment by Martin-Nyaga** [comment]
Hi, I'm glad you'd like to give this a go!

It looks like this is related to your c++ compiler version. What is the output of `g++ --version`? I believe you need to have `gcc` version 8 or higher to compile.

**Comment by vikaswadhwa-dgtl-factory** [comment]
My output is:

```
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/4.2.1
Apple clang version 12.0.5 (clang-1205.0.22.11)
Target: x86_64-apple-darwin20.5.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

**Comment by Martin-Nyaga** [comment]
Okay, I see. I've pushed a potential fix to this branch `fix-clang-compilation`. Can you try and test with this branch and see if it works? If you're using bundler and a Gemfile, here's the snippet to do that:

```
gem 'fast_statistics', github: 'Martin-Nyaga/fast_statistics', branch: 'fix-clang-compilation'
```

**Comment by vikaswadhwa-dgtl-factory** [comment]
Yes that seems to have done it, I'll play around with it some and let you know if there's any issues, thanks so much!

**Comment by Martin-Nyaga** [comment]
Fantastic. I will release a patched version shortly. Please feel free to file another issue if you run into any other problems.

**Comment by Martin-Nyaga** [comment]
This is now solved in version 0.2.1, you can now use this reference instead of the branch.
